dotnet-test-samples/async-architectures/async-lambda-sqs/tests/S3Notifications.E2ETests/FunctionTest.cs

```diff
@@ -0,0 +1,44 @@
+using System.Text.Json;
+using Amazon.S3;
+using Amazon.S3.Model;
+using S3Notifications.integrationTests.Fixtures;
+using Xunit;
+
+namespace S3Notifications.E2ETests;
+
+public class FunctionTest : 
+    IClassFixture<S3NotificationEnvironmentFixture>
+{
+    private readonly S3NotificationEnvironmentFixture _fixture;
+    private readonly IAmazonS3 _s3Client;
+    private readonly string _bucketName;
+
+    public FunctionTest(S3NotificationEnvironmentFixture fixture)
+    {
+        _fixture = fixture;
+        _s3Client = fixture.S3Client;
+        _bucketName = fixture.BucketName;
+    }
+
+    [Fact]
+    public async Task TestS3EventLambdaFunction()
+    {
+        var key = $"text_{Guid.NewGuid()}.txt";
+
+        // Create a bucket an object to setup a test data.
+        await _s3Client.PutBucketAsync(_bucketName);
+        await _s3Client.PutObjectAsync(new PutObjectRequest
+        {
+            BucketName = _bucketName,
+            Key = key,
+            ContentBody = "sample data"
+        });
+
+        var message = await _fixture.GetNextMessage();

```

Comment:
What if there are multiple messages on the queue? How would you manage only getting the one you are interested in?

Reply:
Good question, in case of E2E tests we do not have control on the message itself, so I'll look into an "assertion" that checks for a specific message

Reply:
Added search for specific message type